### PR TITLE
[ci-app] Add HTTP Integrations Tests to Testing Frameworks Instrumentations

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
* Add `http` integration tests to `mocha`, `jest` and `cucumber`

### Motivation
* Improve confidence, as part of the production readiness efforts. 

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/datadog-plugin-cucumber/src/index.js [hunk 1]
```diff
@@ -52,6 +52,10 @@ function createWrapRun (tracer, testEnvironmentMetadata, sourceRoot) {
           const promise = run.apply(this, arguments)
           promise.then(() => {
             setStatusFromResult(testSpan, this.getWorstStepResult(), TEST_STATUS)
+          }).finally(() => {
+            testSpan.context()._trace.started.forEach((span) => {
+              span.finish()
```
Comment (juan-fernandez) [inline packages/datadog-plugin-cucumber/src/index.js:57]: this is in the other testing framework instrumentation but was not in this one. I found out when adding the http integration test

File: packages/datadog-plugin-jest/test/index.spec.js [hunk 2]
```diff
@@ -198,17 +204,20 @@ describe('Plugin', () => {
...
         if (process.env.DD_CONTEXT_PROPAGATION === 'false') return
+        const originalWrap = tracer._tracer.wrap
         tracer._tracer.wrap = sinon.spy(() => {})
 
         const testFnStartEvent = {
           name: 'test_fn_start'
         }
         datadogJestEnv.handleTestEvent(testFnStartEvent)
         expect(tracer._tracer.wrap).not.to.have.been.called
+        tracer._tracer.wrap = originalWrap
```
Comment (juan-fernandez) [inline packages/datadog-plugin-jest/test/index.spec.js:220]: unrelated fix

File: packages/datadog-plugin-cucumber/src/index.js (hunk 1 above)
Comment (rochdev) [reply packages/datadog-plugin-cucumber/src/index.js:57]: Should this be moved to some common wrapper then instead of copying the logic? Especially since eventually there will be a better way to handle this which would otherwise require updating every plugin doing this.

File: packages/datadog-plugin-cucumber/test/index.spec.js [hunk 3]
```diff
@@ -125,13 +146,16 @@ describe('Plugin', () => {
           errors
         } = test
 
-        describe(`for ${featureName}${featureSuffix}`, () => {
-          it('should create a test span', async () => {
+        describe(`for ${featureName}${featureLineNumber}`, () => {
+          it('should create a test span and spans for integrations', async function () {
+            this.timeout(200000)
```
Comment (rochdev) [inline packages/datadog-plugin-cucumber/test/index.spec.js:151]: Why such a high timeout?

File: packages/datadog-plugin-jest/test/index.spec.js [hunk 4]
```diff
@@ -452,6 +461,34 @@ describe('Plugin', () => {
         passingTestEvent.test.fn()
       })
 
+      it('works with http integration', (done) => {
```
Comment (rochdev) [inline packages/datadog-plugin-jest/test/index.spec.js:464]: Is this what is actually tested? Or is it testing context propagation more generally?

Comment (rochdev) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: Also, if it is actually testing propagation more generally, HTTP is not needed which would avoid having to use Nock.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/datadog-plugin-cucumber/src/index.js (hunk 1 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-cucumber/src/index.js:57]: yeah that's a good point. I'll do it

File: packages/datadog-plugin-cucumber/test/index.spec.js (hunk 3 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-cucumber/test/index.spec.js:151]: this is a left over from when I was testing. It shouldn't be needed. I'll remove it

File: packages/datadog-plugin-jest/test/index.spec.js (hunk 4 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: yeah, this is the actual test. 

Do you mean I could use a different library for testing propagation? 
The reason we've singled out http and database integrations is because 
1. they're common
2. http requests can create traces in different services, and it is important that the origin tag is propagated

Comment (rochdev) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: Makes sense for HTTP requests. But then what is additionally tested by #1410 ?

Comment (juan-fernandez) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: #1410 is not ready for review yet. I wanted to give a more thorough description before moving it to ready for review. 

The choice of `pg` as a "database library" is arbitrary in #1410. The one thing that could be interesting in that PR is the ability to test "cross integration", that is, that if a library that we instrument calls another one, things like `origin` are propagated correctly. 
This is easy to test with `http` but was more difficult with `pg`, hence the `TEST_DEPENDENCIES` env var: https://github.com/DataDog/dd-trace-js/pull/1410/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R625. 
I'm not sure if we want to do that though. 

As I mentioned, I wanted to finish up this PR before moving to #1410

Comment (rochdev) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: `pg` wouldn't propagate since it would never have children.

Comment (juan-fernandez) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: yeah that's a good point. We might not need this

File: packages/datadog-plugin-cucumber/test/index.spec.js (hunk 3 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-cucumber/test/index.spec.js]: removed in https://github.com/DataDog/dd-trace-js/pull/1406/commits/e1a029248ba60bcfa057a7ca7cd20c8f870decca

File: packages/datadog-plugin-jest/test/index.spec.js (hunk 2 above)
Comment (rochdev) [reply packages/datadog-plugin-jest/test/index.spec.js:220]: We should probably figure out a way to not do that as soon as possible since it makes assumption on the tracer internals.

**Review by rochdev — APPROVED** [review]

File: packages/datadog-plugin-jest/test/index.spec.js (hunk 2 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-jest/test/index.spec.js:220]: I just created a jira issue to track it

File: packages/datadog-plugin-jest/test/index.spec.js (hunk 4 above)
Comment (juan-fernandez) [reply packages/datadog-plugin-jest/test/index.spec.js:464]: and closed :)
